### ISSUE TITLE
netbox_interface: vlan mode access and name as a number shows false-positive change

### ISSUE THREAD

**Issue by TerraTalpi** [netbox-community/ansible_modules#164]
##### ISSUE TYPE
 - Bug Report

###### Ansible:
```
ansible 2.9.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/dist-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.9 (default, Nov  7 2019, 10:44:02) [GCC 8.3.0]

pip:
pynetbox 4.3.1
```

###### Netbox:
v2.7.10

###### Collection:
v0.1.10

##### SUMMARY
When I execute a play twice with netbox_interface and set the VLAN mode to access and the name to "1" I expect no change to occur. Actually something is changed. It seems to be a false-positive change. Technically the module is still idempotent because the end result is the same, but it shows a change that obviously changes nothing.

##### STEPS TO REPRODUCE
I iterate over an inventory and configure ports of a switch in netbox. I create the Ports named 1,2,3,4 with the play below and configure it as access ports. I expect that no changes occur when I run this play a second time. 

```yaml
    - name: "configure acccess ports on hp switch"
      netbox_interface:
        netbox_url: "{{ netbox_url }}"
        netbox_token: "{{ netbox_token }}"
        data:
          name: "{{ item }}"
          device: "{{ inventory_hostname }}"
          type: 1000base-t
          description: "client access port"
          mode: access
          untagged_vlan: 1
          mtu: 1500
        state: present
      delegate_to: localhost
      loop: "{{ range(1, 5) | list }}"
```

##### EXPECTED RESULTS
No changes in ansible when I execute the play a second time, green result.

##### ACTUAL RESULTS
change occoured, yellow result. Debug output with --vvvv:
```yaml
changed: [myswitch -> localhost] => (item=1) => {
    "ansible_loop_var": "item",
    "changed": true,
    "diff": {
        "after": {
            "mode": 100,
            "name": 1
        },
        "before": {
            "mode": "access",
            "name": "1"
        }
    },
    "interface": {
        ... rest omitted
```
I see two problems here:
1. Obviously the name is rendered as an integer value upon second execution and not as a string. Netbox saves it as a string because it could also be sth like "1/1/1" which is obviously not an int. Should I add a type-cast with `| str` here to prevent this or should this do the netbox_interface module?
1. Mode is being set to "access", but somewhere the mode is converted to int 100 before it is sent to netbox.

**Comment by FragmentedPacket** [comment]
Can you try updating pynetbox to see if they resolves your issue. If not, what version are you at?

**Comment by TerraTalpi** [comment]
I have checked my pynetbox version, it's 4.3.0. I have upgraded to pynetbox 4.3.1. Still the very same behaviour.

**Comment by FragmentedPacket** [comment]
Ok thanks for checking that. I should have noticed this from the beginning, but that is a module I had to move over from Ansible core and isn't supported any longer.

Do you mind trying `netbox_device_interface`?

**Comment by TerraTalpi** [comment]
Oh, I didn't see that the module is deprecated. I checked the readme now, it's not even listed there 😄. I migrated from the standard community netbox library, therefore I did this mistake.

I have now migrated all `netbox_interface` to `netbox_device_interface` and slightly adjusted the parameters to match the new options. Everything works perfectly and there are no more false-positives. Thank you very much!
